lib/std/rand.zig

```diff
@@ -47,6 +47,30 @@ pub const Random = struct {
         return r.int(u1) != 0;
     }
 
+    /// Returns a random value from an enum, evenly distributed.
+    pub fn enumValue(r: *Random, comptime EnumType: type) EnumType {
+        if (comptime !std.meta.trait.is(.Enum)(EnumType)) {
+            @compileError("Random.enumValue requires an enum type, not a "++@typeName(EnumType));
+        }
+
+        // We won't use int -> enum casting because enum elements can have
+        //  arbitrary values.  Instead we'll randomly pick one of the type's
+        //  fields (values).
+        const options = comptime std.meta.fields(EnumType);
+        if (options.len == 0) {
```

Comment:
This check does not need to exist as empty enums are not a thing.

Reply:
Doh, for some reason I thought the spec allowed them.  Must have confused with empty structs.